### DISCUSSION TITLE
Deploying Express/Node backend

### DISCUSSION THREAD

**Discussion by st3fus** [vercel/community#2430] [Help]
### Summary

So I'm trying to deploy a Express/Node backend for my frontend that's also hosted on Vercel. Frontend works fine but it's not communicating properly with the backend, my guess is that Backend isn't getting deployed properly. I have no idea what to do more to make it work, any insight could be helpful! Thanks

### Additional information

This is the current error I'm getting on backend once I go to the deployment:
```
This Serverless Function has crashed.

Your connection is working correctly.

Vercel is working correctly.

500: INTERNAL_SERVER_ERROR
Code: FUNCTION_INVOCATION_FAILED
ID: cdg1::bhr54-1683888166238-ab87b2fe09a5
```

This is the vercel.json i'm currently using:
```
{
	"version": 2,
	"public": true,
	"builds": [
		{
			"src": "src/server.js",
			"use": "@vercel/node"
		}
	],
	"rewrites": [
		{
			"source": "/(.*)",
			"destination": "/src/server.js"
		}
	],
	"headers": [
		{
			"source": "/api/(.*)",
			"headers": [
				{ "key": "Access-Control-Allow-Credentials", "value": "true" },
				{ "key": "Access-Control-Allow-Origin", "value": "*" },
				{ "key": "Access-Control-Allow-Methods", "value": "GET,OPTIONS,PATCH,DELETE,POST,PUT" },
				{
					"key": "Access-Control-Allow-Headers",
					"value": "X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version"
				}
			]
		}
	]
}
```

### Example

https://eko-servis-ns-admin.vercel.app/

**Comment by amyegan** [comment]
Hi @st3fus! Are the front end and backend being deployed to the same project or as two separate projects? If you have a public repo or minimal reproduction you can share, that would help me see get a better look at what's going wrong.

**Reply by eaconcepcion** [reply]
i'm getting the same error. I have client and server folders and deployed separately. The client page can't login. The deployed server part is getting 500 or 404. Please help.  It has been an issue seems no solution or know how.

**Reply by eaconcepcion** [reply]
This is the server part - https://pasaapp.vercel.app/ with 404 error now. Here is the client https://pasaappclient.vercel.app/. The client works but can't login. Many thanks Amy Egan.

**Reply by amyegan** [reply]
You should see more details about the server errors in that project's [runtime logs](https://vercel.com/docs/concepts/observability/runtime-logs). That will help you discover the cause of the error.

You can also check the deployment output by going to deployment summary page in your project dashboard, clicking the `Source` tab, and then toggling from `Source` to `Output` near the top of the page. 

<img width="733" alt="deployment details source tab showing output structure" src="https://github.com/vercel/community/assets/8247105/38cac704-c664-4ea5-961f-f4ff3f801950">

**Comment by st3fus** [comment]
I acutally figured out whats causing the issue on my end with backend. It's the AdminBro I'm using for admin panel, I will switch back to Django and try it again
